commit 349fbfe9b918e6dea00734f07c0fbaf4c2e2df5e
Author: Fabiano Rosas <farosas@linux.ibm.com>
Date:   Tue Jan 25 18:56:54 2022 -0300

    KVM: PPC: mmio: Return to guest after emulation failure
    
    If MMIO emulation fails we don't want to crash the whole guest by
    returning to userspace.
    
    The original commit bbf45ba57eae ("KVM: ppc: PowerPC 440 KVM
    implementation") added a todo:
    
      /* XXX Deliver Program interrupt to guest. */
    
    and later the commit d69614a295ae ("KVM: PPC: Separate loadstore
    emulation from priv emulation") added the Program interrupt injection
    but in another file, so I'm assuming it was missed that this block
    needed to be altered.
    
    Also change the message to a ratelimited one since we're letting the
    guest run and it could flood the host logs.
    
    Signed-off-by: Fabiano Rosas <farosas@linux.ibm.com>
    Reviewed-by: Nicholas Piggin <npiggin@gmail.com>
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Link: https://lore.kernel.org/r/20220125215655.1026224-5-farosas@linux.ibm.com

diff --git a/arch/powerpc/kvm/powerpc.c b/arch/powerpc/kvm/powerpc.c
index 27fb2b70f631..acb0d2a4bdb9 100644
--- a/arch/powerpc/kvm/powerpc.c
+++ b/arch/powerpc/kvm/powerpc.c
@@ -307,9 +307,9 @@ int kvmppc_emulate_mmio(struct kvm_vcpu *vcpu)
 		u32 last_inst;
 
 		kvmppc_get_last_inst(vcpu, INST_GENERIC, &last_inst);
-		/* XXX Deliver Program interrupt to guest. */
-		pr_emerg("%s: emulation failed (%08x)\n", __func__, last_inst);
-		r = RESUME_HOST;
+		kvm_debug_ratelimited("Guest access to device memory using unsupported instruction (opcode: %#08x)\n",
+				      last_inst);
+		r = RESUME_GUEST;
 		break;
 	}
 	default: